packages/quantic/force-app/main/default/lwc/quanticResultsPerPage/e2e/quanticResultsPerPage.e2e.ts

```diff
@@ -0,0 +1,97 @@
+import {testSearch, testInsight} from './fixture';
+import {
+  useCaseEnum,
+  useCaseTestCases,
+} from '../../../../../../playwright/utils/useCase';
+
+const fixtures = {
+  search: testSearch,
+  insight: testInsight,
+};
+
+const defaultChoices = [10, 25, 50, 100];
+const defaultInitialChoice = 10;
+
+useCaseTestCases.forEach((useCase) => {
+  let test = fixtures[useCase.value];
+
+  test.describe(`quantic results per page ${useCase.label}`, () => {
+    test.describe('with default options', () => {
+      test('should load as expected', async ({resultsPerPage}) => {
+        await test.expect(resultsPerPage.component).toBeVisible();
+        test
+          .expect(await resultsPerPage.allResultsPerPageOptionsButtons.count())
+          .toBe(defaultChoices.length);
+        test
+          .expect(resultsPerPage.selectedResultsPerPageOption)
+          .toHaveText(defaultInitialChoice.toString());
+      });
+
+      test.describe('when selecting a new results per page option', () => {
+        test('should trigger a new search and log analytics', async ({
+          resultsPerPage,
+          search,
+        }) => {
+          const newChoice = 25;
+
+          const searchResponsePromise = search.waitForSearchResponse();
+          const resultsPerPageAnalyticsPromise =
+            resultsPerPage.waitForResultsPerPageUA({
+              currentResultsPerPage: newChoice,
+            });
+
+          await resultsPerPage.clickResultsPerPageOptionByValue(newChoice);
+
+          const searchResponse = await searchResponsePromise;
+          await resultsPerPageAnalyticsPromise;
+
+          test
+            .expect(resultsPerPage.selectedResultsPerPageOption)
+            .toHaveText(newChoice.toString());
+          test
+            .expect(searchResponse.request().postDataJSON()?.numberOfResults)
+            .toBe(newChoice);
+        });
+      });
+    });
+
+    test.describe('with custom options', () => {
+      const customChoices = [20, 40, 80];
+      test.use({
+        options: {
+          choicesDisplayed: customChoices.join(','),
+        },
+      });
+
+      test('should load as expected', async ({resultsPerPage}) => {
+        await test.expect(resultsPerPage.component).toBeVisible();
+        test
+          .expect(await resultsPerPage.allResultsPerPageOptionsButtons.count())
+          .toBe(customChoices.length);
+      });
```

Comment:
Same remark here, for me this is a unit test that you already did.

Reply:
See reply https://github.com/coveo/ui-kit/pull/4889#discussion_r1932392914